session.go

```diff
@@ -947,6 +956,44 @@ func (q *Query) MapScanCAS(dest map[string]interface{}) (applied bool, err error
 	return applied, iter.Close()
 }
 
+// Release releases a query back into a pool of queries. Released Queries
+// cannot be reused.
+//
+// Example:
+// 		qry := session.Query("SELECT * FROM my_table")
+// 		qry.Exec()
+// 		qry.Release()
+func (q *Query) Release() {
+	if q == nil || q.session == nil || q.session.queryPool == nil {
```

Comment:
this can be simplified if the pool is a package local variable then it will never be nil, how worthwhile is the query nil check?